### DISCUSSION TITLE
Problem with APIFlask generating docs with flask_apscheduler extension.

### DISCUSSION THREAD

**Discussion by diten** [apiflask/apiflask#206] [Q&A] [answered]
Hi, 
I'm pretty new to APIFlask, but I'm starting to like it, a lot. So really great work you guys/girls.
I think there must be a better way to fix this, I just tell what I have done meanwhile.

I'm facing a small problem, which I sort of fix it, but I just comment it here so if is not anywhere (I can't find anything about it), it's helpful to others.

I had a small app, I converted it to APIFlask, but it uses APScheduler, and it keep crashing.
If I remove APScheduler, everything worked right.
So with a little debugging I got to the culprit.

In `app.py._generate_spec` : 
when it iterates over the rules, the scheduler rules aren't  well managed and crash the spec generation.
(I don't quite get if APScheduler definitions are right o wrong).

So my fix was really fast and ugly, but again, it could help somebody else:

Just before `for rule in rules:` I filtered all the /scheduler rules 
```
        # extract scheduler rules as  a temp fix.
        rules: t.List[t.Any] = list(filter(lambda x: not str(x).startswith('/scheduler'), rules))
```

This let me keep using APIFlask and APscheduler altogether.
Scheduler functionality is obviously not in OAS Specs, but I have tried using `@doc(hide=True)` over the Scheduler definition to try to hide it, first, cause I don't want to show it, and then to see if that help solve the problem, but it didn't worked out.
Nonetheless, Scheduler functionality is there and working perfectly now.

Bye, cheers, Diten.

**Comment by greyli** [comment, marked as answer]
Hi, thanks for reporting this! I will try to reproduce and fix it this weekend, then make a new fix release.

Just created a new issue from this: #211
